docs/ossm/ossm2-migration/cluster-wide/README.md

```diff
@@ -0,0 +1,96 @@
+> [!CAUTION]
+> In case you decide to use a `default` name for 3.0 `Istio` resource with `InPlace` upgrade strategy, 3.0 control plane will try to inject side cars to all pods in namespaces with `istio-injection=enabled` label and all pods with `sidecar.istio.io/inject="true"` label after next restart of the workloads. Without using `maistra.io/ignore-namespace: "true"` label, it will result in a conflict with 2.6 control plane injector.
+
+In case you are using `istio-injection=enabled` label on your data plane namespaces and planning to keep it, you should first read revision tag documentation. TODO: add link. TODO: describe cases where the migration will work with `istio-injection=enabled` (default revision, default revision tag)
+
+
+#### Install OpenShift Service Mesh 3
+
+1. Create your `Istio` resource.
+
+    Here we are not using any `discoverySelectors` so the control plane will have access to all namespaces. In case you want to define `discoverySelectors`, keep in mind that all data plane namespaces you are planning to migrate must be matched.
+    > [!CAUTION]
+    > it is important your `Istio` resource's `spec.namespace` field is the **same** namespace as your `ServiceMeshControlPlane`. If you set your `Istio` resource's `spec.namespace` field to a different namespace than your `ServiceMeshControlPlane`, the migration will not work properly.
+    ```yaml
+   apiVersion: sailoperator.io/v1alpha1
+   kind: Istio
+   metadata:
+     name: ossm-3
+   spec:
+     namespace: istio-system
+     version: v1.23.2
+   ```
+
+#### Migrate Workloads
+
+1. Update injection labels on the data plane namespace.
+
+    Here we're adding two labels to the namespace:
+
+    1. The `istio.io/rev: ossm-3` label which ensures that any new pods that get created in that namespace will connect to the 3.0 proxy.
+    2. The `maistra.io/ignore-namespace: "true"` label which will disable sidecar injection for 2.6 proxies in the namespace. This ensures that 2.6 will stop injecting proxies in this namespace and any new proxies will be injected by the 3.0 control plane. Without this, the 2.6 injection webhook will try to inject the pod and it will connect to the 2.6 proxy as well as refuse to start since it will have the 2.6 cni annotation.
+
+    > [!NOTE]
+    > that once you apply the `maistra.io/ignore-namespace` label, any new pod that gets created in the namespace will be connected to the 3.0 proxy. Workloads will still be able to communicate with each other though regardless of which control plane they are connected to.
+
+    ```sh
+    oc label ns bookinfo istio.io/rev=ossm-3 maistra.io/ignore-namespace="true" --overwrite=true
+    ```
+    > [!IMPORTANT]
+    > In case you are using `istio-injection=enabled` label on the namespace being migrated, remove that label via `oc label ns bookinfo istio-injection-` or add a default revision tag. TODO: add more when the revision tag doc is ready
```

Comment:
If you are using `istio-injection=enabled` you wouldn't need to add `istio.io/rev` at all if you had a revision tag deployed.

What's the desired end state? Users have `istio-injection=enabled` label or `istio.io/rev` label? Or is it just user preference and they can use either one? If users want to keep `istio-injection=enabled`, we may need to have separate instructions here depending on which injection label they want.

Reply:
Agree, this is confusing and mixing two different scenarios. Adding separated instructions for `istio-injection=enabled` and `istio.io/rev` will be clearer.